auacm/app/modules/competition_manager/views.py

```diff
@@ -181,3 +202,149 @@ def get_competition_data(cid):
+
+    All the user has to do is submit a post to this url with no form data.
+    From their logged-in status, we'll go ahead and add them to the competiton
+    as an individual (team name is default their display name). A 400 error will
+    be returned if the user is already registered for the competition.
+
+    If the user that is submitting this is an admin, they can optionally
+    supply a json array of usernames to register for the competition. Specifying
+    this will not register the admin, but it will register all users that are
+    listed. A 400 error will be returned if any of the users are already
+    registered for the competition.
+    """
+    if session.query(Competition).filter(Competition.cid == cid).first() \
+            is None:
+        return serve_error('Competition does not exist', response_code=404)
+
+    if current_user.admin == 1 and 'users' in request.data:
+        registrants = loads(request.data['users'])
+    else:
+        registrants = list()
+        registrants.append(current_user.username)
+
+    for user in registrants:
+        if session.query(CompUser).filter(CompUser.cid == cid,
+            CompUser.username == user).first() is not None:
+            return serve_error('User ' + user + ' already registered for '
+                    'competition', response_code=400)
+
+    for username in registrants:
+        user = session.query(User).filter(User.username == user).first()
+        session.add(CompUser(
+            cid=cid,
+            username=user.username,
+            team=user.display
+        ))
+        socketio.emit('new_user', {
+            'cid': cid,
+            'user': {
+                'display': user.display,
+                'username': user.username
+            }
+        },
+        namespace='/register')
+    session.flush()
+    session.commit()
+
+    return serve_response({})
+
+
+@app.route('/api/competitions/<int:cid>/unregister', methods=['POST'])
+@login_required
+def unregister_for_competition(cid):
+    """ Called when a user wants to register for a competition.
+
+    All the user has to do is submit a post to this url with no form data.
+    From their logged-in status, we'll go ahead and remove them from the
+    competiton.
+
+    Similar to the <code>/register</code> endpoint, an admin can post a list of
+    users to unregister from the competition.
+    """
+    if session.query(Competition).filter(Competition.cid == cid).first() \
+            is None:
+        return serve_error('Competition does not exist', response_code=404)
+
+    if current_user.admin == 1 and 'users' in request.data:
+        registrants = loads(request.data['users'])
+    else:
+        registrants = list()
+        registrants.append(current_user.username)
+
+    for user in registrants:
+        (session.query(CompUser)
+                .filter(CompUser.username == user, CompUser.cid == cid)
+                .delete())
```

Comment:
I like the parens over backslash. I'm pretty sure that follows Google style, despite some of our legacy code. We should move forward with this and (at some point) change the old ones.